bin/varnishd/cache/cache_fetch.c

```diff
@@ -532,6 +532,10 @@ vbf_stp_startfetch(struct worker *wrk, struct busyobj *bo)
 		bo->uncacheable = 1;
 		wrk->handling = VCL_RET_DELIVER;
 	}
+	if (bo->do_pass && !bo->do_stream) {
+		/* Pass prefetch should be disabled when not streaming. */
+		bo->fetch_objcore->boc->pass_prefetch = 0;
+	}
```

Comment:
I think the condition should actually be `if (!bo->uncacheable || !bo->do_stream)` since `bo->do_pass` is off-topic on account of belonging to `bereq` and we need both to let `pass_prefetch` kick in.